runtime/src/cost_model.rs

```diff
@@ -126,9 +126,18 @@ impl CostModel {
         // by `bank`, therefore it should be considered as no execution cost by cost model.
         match result {
             Ok(_) => {
-                // if tx contained user-space instructions and a more accurate estimate available correct it
-                if bpf_costs > 0 {
-                    bpf_costs = compute_budget.compute_unit_limit
+                if feature_set.is_active(&native_programs_consume_cu::id()) {
+                    // if feature gate is activated, the requested compute_unit_limit covers both
+                    // builtin and bpf programs
+                    builtin_costs = compute_budget.compute_unit_limit.min(builtin_costs);
+                    bpf_costs = compute_budget
+                        .compute_unit_limit
+                        .saturating_sub(builtin_costs);
+                } else {
+                    // if tx contained user-space instructions and a more accurate estimate available correct it
+                    if bpf_costs > 0 {
```

Comment:
why do we need to check `bpf_costs > 0` here but not in the other branch

Reply:
In this `else` branch, it _only_ needs to take user requested compute_unit_limit as bpf_execution_cost when there is bpf ix in tx (determined by `bpf_cost>0`); in the other branch, requested compute_unit_limit (its value could be user specified or default value) is for both builtin and bpf, so it is _always_ the total programs_execution_cost. 

After `native_programs_consume_cu` is enabled, a follow up will be to remove `builtins_executio_cost` and `bpf_execution_cost`, replace with `programs_execution_cost`(which I have already added)